### PR TITLE
Change to single drift chamber and modify the percentage of gas elements

### PR DESCRIPTION
**Pull request by myliu-hub**
Change the dual drift chamber to a single drift chamber, where
1. inner radius:800mm
2. outer radius:1720mm
The radii of the inner and outer walls respectively extend outward.
  Due to the modification of the inner and outer radius of the drift chamber, the parameters of the silicon detector should also be corrected accordingly. Since I am not clear, I added the parameters of the single drift chamber in the `Detector/DetCRD/compact/CRD_o1_v01/CRD_Dimensions_v01_01.xml` and `Detector/DetCRD/compact/CRD_o1_v02/CRD_Dimensions_v01_02.xml` , and the parameters of the dual drift chamber were not deleted before.

### CODE REVIEW

**Comment by myliu-hub** [comment]
@fucd ,Would you please check if this is reasonable?

**Comment by fucd** [comment]
@myliu-hub Do you decide to replace the cpp file not create a new one with version number?

File: Detector/DetCRD/compact/CRD_common_v01/DC_Simple_v01_01.xml [hunk 1]
```diff
@@ -11,41 +13,39 @@
   </info>
 
   <define>
-    <constant name="SDT_radius_min" value="InnerTracker_inner_radius"/>
-    <constant name="SDT_radius_max" value="OuterTracker_outer_radius"/>
+    <constant name="world_size" value="2226*mm"/>
```
Comment (fucd) [inline Detector/DetCRD/compact/CRD_common_v01/DC_Simple_v01_01.xml:16]: not used? And world should not be defined in sub-detector option file.

File: Detector/DetCRD/compact/CRD_common_v01/DC_Simple_v01_01.xml [hunk 2]
```diff
@@ -11,41 +13,39 @@
...
+    <constant name="world_x" value="world_size"/>
+    <constant name="world_y" value="world_size"/>
+    <constant name="world_z" value="world_size"/>
+
+    <!-- SDT -->
+    <constant name="DetID_DC"  value="7"/>
```
Comment (fucd) [inline Detector/DetCRD/compact/CRD_common_v01/DC_Simple_v01_01.xml:22]: has defined in CRD_Dimensions_v01_01.xml

File: Detector/DetCRD/compact/CRD_o1_v01/CRD_Dimensions_v01_01.xml [hunk 3]
```diff
@@ -4,11 +4,11 @@
        xs:noNamespaceSchemaLocation="http://www.lcsim.org/schemas/compact/1.0/compact.xsd">
 
   <info name="CRDDimensions"
-	title="master file with includes and world dimension"
-	author="C.D.Fu, "
-	url="no"
-	status="development"
-	version="1.0">
+       title="master file with includes and world dimension"
+       author="C.D.Fu, "
```
Comment (fucd) [inline Detector/DetCRD/compact/CRD_o1_v01/CRD_Dimensions_v01_01.xml:8]: any contributor can be added here

File: Detector/DetCRD/compact/CRD_o1_v01/CRD_Dimensions_v01_01.xml [hunk 4]
```diff
@@ -81,27 +81,28 @@
...
-    
+    <constant name="OuterTracker_inner_radius" value="1082.18*mm"/>
+    <constant name="OuterTracker_outer_radius" value="1723*mm"/>
+
+    <!-- Parameters of single drift chamber  -->
+    <constant name="DC_chamber_layer_rbegin" value="800*mm"/>
+    <constant name="DC_chamber_layer_rend" value="1720*mm"/>
+    <constant name="DC_chamber_radius_min" value="DC_chamber_layer_rbegin-DC_safe_distance"/>
+    <constant name="DC_chamber_radius_max" value="DC_chamber_layer_rend+DC_safe_distance"/>
+    <constant name="Tracker_half_length"  value="DC_half_length" />
```
Comment (fucd) [inline Detector/DetCRD/compact/CRD_o1_v01/CRD_Dimensions_v01_01.xml:102]: Tracker is a prefix easy to confuse. Can you rename it?

**Comment by myliu-hub** [comment]
Change the dual drift chamber to a single drift chamber, where

inner radius:800mm
outer radius:1720mm

@fucd， I made some modifications based on your suggestions. At the same time, due to the change of the radius, there will be an overlap with the silicon detector. You may need to update the silicon detector later.

**Comment by myliu-hub** [comment]
Sorry, the outer radius is 1800mm

File: Detector/DetDriftChamber/compact/det.xml [hunk 5]
```diff
@@ -26,8 +26,8 @@
 
     <!-- SDT -->
     <constant name="DetID_DC"  value="7"/>
-    <constant name="SDT_radius_min" value="234*mm"/>
-    <constant name="SDT_radius_max" value="1720*mm"/>
+    <constant name="SDT_radius_min" value="799.78*mm"/>
+    <constant name="SDT_radius_max" value="1803*mm"/>
```
Comment (mirguest) [inline Detector/DetDriftChamber/compact/det.xml:30]: Hi @myliu-hub , I think these value should not be hardcoded.

Tao

File: Detector/DetDriftChamber/src/driftchamber/DriftChamber.cpp [hunk 6]
```diff
@@ -237,34 +204,24 @@ static dd4hep::Ref_t create_detector(dd4hep::Detector& theDetector,
...
 
-         det_inner_chamber_phy.addPhysVolID("chamber", 0);
-    }
-    // outer
-    dd4hep::Transform3D transform_outer_chamber(dd4hep::Rotation3D(),
-            dd4hep::Position(0,0,0));
-    if(outer_chamber_enabled) {
-       dd4hep::PlacedVolume det_outer_chamber_phy = det_vol.placeVolume(det_outer_chamber_vol,
-               transform_inner_chamber);
+    det_chamber_phy.addPhysVolID("chamber", 0);
```
Comment (mirguest) [inline Detector/DetDriftChamber/src/driftchamber/DriftChamber.cpp:216]: For the chamber id, you should avoid the hardcoded number. 
Maybe the hardware groups need the multiple chambers in the future. 

Tao

**Review by mirguest — CHANGES_REQUESTED** [review]
Hi @myliu-hub , please have a look at my comments. 
Thank you.
Tao

File: Detector/DetCRD/compact/CRD_common_v01/DC_Simple_v01_01.xml [hunk 7]
```diff
@@ -61,40 +56,34 @@
   </regions>
 
   <detectors>
-    <detector id="DetID_DC" name="DriftChamber" type="DriftChamber" readout="DriftChamberHitsCollection" vis="BlueVis" sensitive="true" insideTrackingVolume="true" limits="DC_limits">
+    <detector id="DetID_DC" name="DriftChamber" type="DriftChamber" readout="DriftChamberHitsCollection" vis="VisibleBlue" sensitive="true" region="DriftChamberRegion" limits="DC_limits">
       <envelope vis="SeeThrough">
```
Comment (mirguest) [inline Detector/DetCRD/compact/CRD_common_v01/DC_Simple_v01_01.xml:60]: Hi @myliu-hub ,
As you already add the chamber id, do you need to update this part? Or you already define the default value?
Tao

**Review by mirguest — CHANGES_REQUESTED** [review]

**Review by mirguest — APPROVED** [review]

**Pull request merged**